### PR TITLE
Okta + Duo: Allow passcodes from mobile app

### PR DESCRIPTION
**Pull request by wlonkly**
Any Duo mobile app registered with Duo, usually for Duo Push, also has the capability to produce HOTP passcodes:

Duo Mobile app on iOS, showing HOTP code. Yes, I burned that code. 

In the case where Duo Push is administratively disabled and Yubikeys (etc) are not provided, HOTP might be the only way for a user to perform Duo MFA. This is the case for us, which is preventing our non-Yubikey users from using saml2aws at all. 

This change adds Passcode to the list of MFA possibilities whenever a `phone1` is registered with Duo, to support using HOTP. 

--- 

I debated between adding it like this, or adding another conditional to the `option[value="token"]` case -- let me know if that would be preferable. I also checked the other Duo-enabled providers, but none of the others use this kind of logic to limit the MFA options presented to the user.

### CODE REVIEW

**Review by mapkon — APPROVED** [review]
Looks fine to me. @gliptak / @wolfeidau - any thoughts?

**Comment by gliptak** [comment]
any related unit test updates?

**Comment by gliptak** [comment]
@wlonky yes, there seems to be coverage the modified code https://app.codecov.io/gh/Versent/saml2aws/blob/master/pkg%2Fprovider%2Fokta%2Fokta.go#L974

consider creating a copy of https://github.com/Versent/saml2aws/blob/14c45abf5f8dcee4d39a269cf1a55dcf28006643/pkg/provider/okta/okta_test.go#L268 and updating for 1Passcode` path (some processing could be shared between these two flows)

**Comment by wlonkly** [comment]
Got it, I'll see what I can do! Probably won't get to this until the weekend.

**Comment by wlonkly** [comment]
Sorted and ready to go!

File: pkg/provider/okta/okta_test.go [hunk 1]
```diff
@@ -266,9 +268,108 @@ func TestVerifyMfa(t *testing.T) {
...
+	loginDetails := &creds.LoginDetails{URL: ts.URL, Username: "user@example.com", Password: "test123"}
+	return ac, loginDetails
+}
+
+func setupTestDuoHttpServer(t *testing.T, duoFactor string) (*httptest.Server) {
 	verifyCounter := 0
 	statusCounter := 0
-	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+  ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```
Comment (gliptak) [inline pkg/provider/okta/okta_test.go:372]: this seems to have wrong indent

Comment (wlonkly) [reply pkg/provider/okta/okta_test.go]: ty, fixed

**Comment by mapkon** [comment]
@gliptak Is this ready to merge?

**Review by mapkon — CHANGES_REQUESTED** [review]
Moving back to needs work since there is an issue with Okta

**Comment by mapkon** [comment]
> Rats, I need to make one more change -- I just used my local build on my actual work Okta account, and I see duplicate "Passcode" options (because I have both `token` and `phone1` authenticators; my test account only had `phone1`).

OK - just let me know when its ready

**Comment by wlonkly** [comment]
@mapkon Should be all set now. I switched around the logic to only add `Passcode` once, rather than uniq-ing it at the end.

**Review by mapkon — APPROVED** [review]

**Pull request merged**